contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-epilogue.y.c

```diff
@@ -374,6 +374,163 @@ tsql_check_param_readonly(const char *paramname, TypeName *typename, bool readon
 	}
 }
 
+Node *
+TsqlOpenJSONSimpleMakeFuncCall(Node* jsonExpr, Node* path)
+{
+    FuncCall *fc;
+    if(path)
+    {
+        fc = makeFuncCall(TsqlSystemFuncName("openjson_simple"), list_make2(jsonExpr, path), COERCE_EXPLICIT_CALL, -1);
+    }
+    else
+    {
+        fc = makeFuncCall(TsqlSystemFuncName("openjson_simple"), list_make1(jsonExpr), COERCE_EXPLICIT_CALL, -1);
+    }
+    return (Node*) fc;
+}
+
+/*
+* This function takes a JsonExpression, path, column list, and optional alias
+* It acts as a bridge between the parser and the json_with function by correctly
+* assembling the function arguments, column definitions list, and alias
+*/
+Node *
+TsqlOpenJSONWithMakeFuncCall(Node* jsonExpr, Node* path, List* cols, Alias* alias)
+{
+    FuncCall *fc;
+    List *jsonWithParams = list_make2(jsonExpr, path);
+    ListCell *lc;
+    RangeFunction *rf = makeNode(RangeFunction);
+    Alias *a = makeNode(Alias);
+    a->aliasname = alias != NULL ? alias->aliasname : "f";
+
+    foreach(lc, cols)
+    {
+        OpenJson_Col_Def *cd = (OpenJson_Col_Def*) lfirst(lc);
+        int initialTmod = getNVarcharTypMod(cd->elemType);
+        ColumnDef *n = (ColumnDef *) createOpenJsonWithColDef(cd->elemName, cd->elemType);
+        StringInfo format_cols = makeStringInfo();
+
+        if(strcmp(cd->elemPath, "") == 0)
+        {
+            // If not path is provided with use the standard path [$.columnName]
+            appendStringInfo(format_cols, "$.%s ", cd->elemName);
+        }
+        else
+        {
+            appendStringInfo(format_cols, "%s ", cd->elemPath);
+        }
+
+        // character types need to have the typmod appended to them
+        if((pg_strcasecmp(TypeNameToString(cd->elemType), "nvarchar") == 0 ||
+            (pg_strcasecmp(TypeNameToString(cd->elemType), "varchar") == 0 ||
+            (pg_strcasecmp(TypeNameToString(cd->elemType), "nchar") == 0 ||
+            pg_strcasecmp(TypeNameToString(cd->elemType), "char") == 0))))
+        {
+            int newTypMod = getNVarcharTypMod(cd->elemType);
+            char buffer [sizeof(int)*8+1];
+            sprintf(buffer, "%d", newTypMod);
```

Comment:
Is there a reason to use `sprintf` instead of `appendStringInfo`?

Reply:
I used it to convert the int to a string. Is it possible to do this with appendStringInfo?